src/librustc_typeck/check/mod.rs

```diff
@@ -4211,11 +4172,8 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
             Def::ForeignMod(..) |
             Def::Local(..) |
             Def::Label(..) |
-            Def::Upvar(..) => {}
-
-            Def::Err => {
-                self.set_tainted_by_errors();
```

Comment:
`Def::Err` never reaches `instantiate_value_path`, it would fail on `def.def_id()`.